File: 2399-check-distances-between-same-letters.py
from typing import List


class Solution:
    def checkDistances(self, s: str, distance: List[int]) -> bool:
        pos = [0] * 26
        for i, ch in enumerate(s):
            idx = ord(ch) - ord('a')
            pos[idx] = i - pos[idx]
        for j in range(26):
            if pos[j] != 0 and pos[j] != distance[j] + 1:
                return False
        return True


solution = Solution()
print(solution.checkDistances(s="abaccb",
                              distance=[1, 3, 0, 5, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]))
print(solution.checkDistances(s="aa",
                              distance=[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]))

File: 2399-check-distances-between-same-letters.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	bool checkDistances(string s, vector<int>& distance) {
		bool visited[26] = { false };
		for (int i = 0; i < s.size(); i++) {
			int idx = s[i] - 'a';
			if (visited[idx]) {
				if (distance[idx] != i) {
					return false;
				}
			}
			else {
				distance[idx] += i + 1;
				visited[idx] = true;
			}
		}
		return true;
	}
};

int main(void) {
	Solution solution = Solution();
	cout << solution.checkDistances("abaccb", vector<int>{1, 3, 0, 5, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}) << endl;
	cout << solution.checkDistances("aa", vector<int>{1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}) << endl;
}